internal/services/elastic/elastic_tag_rule_resource.go

```diff
@@ -0,0 +1,298 @@
+	monitorId, err := parse.ElasticMonitorID(d.Get("monitor_id").(string))
+	if err != nil {
+		return err
+	}
+	ruleSetName := d.Get("name").(string)
+
+	id := parse.NewElasticTagRuleID(subscriptionId, monitorId.ResourceGroup, monitorId.MonitorName, ruleSetName)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, id.ResourceGroup, id.MonitorName, id.TagRuleName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+		}
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_elastic_tag_rule", *existing.ID)
+		}
+	}
+
+	body := elastic.MonitoringTagRules{
+		Properties: &elastic.MonitoringTagRulesProperties{
+			LogRules: expandLogRules(d.Get("log_rules").([]interface{})),
+		},
+	}
+	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.MonitorName, id.TagRuleName, &body); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+	return resourceElasticTagRuleRead(d, meta)
+}
+
+func resourceElasticTagRuleRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Elastic.TagRuleClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.ElasticTagRuleID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.MonitorName, id.TagRuleName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Elastic monitor %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	d.Set("monitor_id", parse.NewElasticMonitorID(id.SubscriptionId, id.ResourceGroup, id.MonitorName).ID())
+	d.Set("name", id.TagRuleName)
+
+	if props := resp.Properties; props != nil {
+		if err := d.Set("log_rules", flattenLogRules(props.LogRules)); err != nil {
+			return fmt.Errorf("setting `log_rules`: %+v", err)
+		}
+	}
+	d.SetId(id.ID())
+
+	return nil
+}
+
+func resourceElasticTagRuleDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Elastic.TagRuleClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err1 := parse.ElasticTagRuleID(d.Id())
+	if err1 != nil {
+		return err1
+	}
+
+	resp, err2 := client.Get(ctx, id.ResourceGroup, id.MonitorName, id.TagRuleName)
+	if err2 != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Elastic %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+	}
+
+	d.Set("log_rules", nil)
+
+	body := elastic.MonitoringTagRules{
+		Properties: &elastic.MonitoringTagRulesProperties{
+			LogRules: expandLogRules(d.Get("log_rules").([]interface{})),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.MonitorName, id.TagRuleName, &body); err != nil {
```

Comment:
I think we should use `client.Delete`?

Reply:
will be updated in the next commit

Reply:
Reference previous conversation [https://github.com/hashicorp/terraform-provider-azurerm/pull/14821#discussion_r795040966](url)